corehq/ex-submodules/casexml/apps/phone/utils.py

```diff
@@ -48,21 +49,26 @@ def get_or_cache_global_fixture(restore_state, cache_bucket_prefix, fixture_name
     domain = restore_state.restore_user.domain
 
     data = None
+    key = '{}/{}'.format(cache_bucket_prefix, domain)
+
     if not restore_state.overwrite_cache:
         data = get_cached_fixture_items(domain, cache_bucket_prefix)
+        _record_datadog_metric('cache_miss' if data is None else 'cache_hit', key)
 
     if data is None:
-        with CriticalSection(['{}/{}'.format(cache_bucket_prefix, domain)]):
-            # re-check cache to avoid re-computing it
-            data = get_cached_fixture_items(domain, cache_bucket_prefix)
-            if data is not None:
-                return [data]
-            else:
-                items = data_fn()
-                io_data = write_fixture_items_to_io(items)
-                data = io_data.read()
-                io_data.seek(0)
-                cache_fixture_items_data(io_data, domain, fixture_name, cache_bucket_prefix)
+        with CriticalSection([key]):
+            if not restore_state.overwrite_cache:
+                # re-check cache to avoid re-computing it
+                data = get_cached_fixture_items(domain, cache_bucket_prefix)
```

Comment:
Does TCL's object storage have eventual consistency, or immediate consistency? I assume that its eventual and that means that this could often return `None` for a short time after its written, which could result in the issues we've seen recently

Reply:
it should be immediate for reads and writes but not for list, update, delete